### DISCUSSION TITLE
How do I enforce lazy loading in a OneToOne relationship?

### DISCUSSION THREAD

**Discussion by jeongsik-woo** [mikro-orm/mikro-orm#5489] [Q&A]
How do I enforce lazy loading in a OneToOne relationship?
The documentation at https://mikro-orm.io/docs/next/relationships#onetoone only gives examples of eager loading, but I want to implement lazy loading.

For example, User owns a Book and you want to lazy load user from the Book entity.
Like this
`await book.user.load();`

Is this possible?

Can you give me an example?

**Comment by B4nan** [comment]
Lazy loading is the default. Why do you think the docs show an example of eager loading? It is not.

**Reply by B4nan** [reply]
You can wrap relations with `Ref` wrapper, if that's what you want - it's optional and you can do the same without it too (with the help of [`wrap` helper](https://mikro-orm.io/docs/entity-helper#wrappedentity-and-wrap-helper) - `wrap(book.user).init()`, although `init` will always reload the values, while `Ref.load` won't do anything on an already loaded relation).

https://mikro-orm.io/docs/type-safe-relations#reference-wrapper

Then you have the `load` method on the relation wrapper.

**Reply by jeongsik-woo** [reply]
When this code executed `const book = await entityManager.findOne(Book, {id: 1})`, then native query is like this.
```
select 
  `c0`.`id`, 
  `c0`.column1, 
  `c0`.column2, 
  `c1`.`id` as `user_id`
from 
  `book` as `c0` 
  left join `user` as `c1` on `c0`.`id` = `c1`.`book_id` 
where `c0`.`id` = '1' 
limit  1;

```

I think it is eager loading. I want to do not using join on findOne. 
We expect the execution to look like this

When I call `findOne` on Book, I want to execute the query only on book, like `select * from book where id =1;.`
And afterward, when I run `await book.user.load()`, I'm looking for user via fk, as shown below.
`select * from user where book_id=1`
Is this possible?

below is my entity configuration.

```
@Entity()
class Book {

  @OneToOne(() => User, (user) => user.book, {
    mappedBy: 'user',
    ref: true,
    lazy: true,
  })
  user: Ref<User>;
  
  ...
}

@Entity()
class User {
  
    @OneToOne(() => Book, { owner: true, ref: true })
  book: Book;
  ....
}
```

**Reply by B4nan** [reply]
Nope, that's not eager loading, this only loads the FK of your relation, nothing else. You won't get a loaded relation from this, only a reference. You can disable this globally https://mikro-orm.io/docs/configuration#auto-join-of-11-owners

But it sounds like you are not using the latest version, since you say you define the property as `lazy: true`, which actually changed in 6.2.1, and we now skip this join in such case 6442e5739fb8e3ed8284ec3bba64c7ee14808b97 

> And afterward, when I run await book.user.load(), I'm looking for user via fk, as shown below.

And that's exactly why you need this join you don't like - without it, `book.user` will be undefined, you won't know the FK, as its an inverse side of the relation. I think you should be still able to populate such relation without the join via `em.populate(book, ['user'])`, but surely not that way via `ref.load()` as the whole entity reference is missing if there is no FK.

**Reply by jeongsik-woo** [reply]
I updated version as 6.2.2 now.
I have one more question.
If I set lazy: true on OneToOne Column, Then I run this code `await book.user.load()`, it occurs error `Cannot read properties of undefined (reading 'load')`. because of book.user is undefined.

Can you see the effect of setting lazy: true?

**Reply by B4nan** [reply]
Yes, that's exactly what I told you in the previous comment. And with `lazy: true` it would happen with an owning side as well, same for m:1 relations. You don't select the FK, so you don't have the value.

**Reply by jeongsik-woo** [reply]
Thank you. When I set lazy: true, loaded entity has not Fk. is it right?
It's probably better not to set lazy: true unless absolutely necessary.
